cirq/sim/simulator.py

```diff
@@ -357,6 +356,19 @@ def _simulator_iterator(
         Yields:
             StepResults from simulating a Moment of the Circuit.
         """
+        param_resolver = param_resolver or study.ParamResolver({})
```

Comment:
The docstring for this method should be expanded to include the details in the previous implementations.

Reply:
Done, but another question is, would it make sense to inline `_simulator_iterator` into `simulate_moment_steps` now? That is the only caller, and it is a thin wrapper. Doing so would also involve updating the mocks in two unit tests.

Reply:
That would be a breaking change, as (I assume) any external implementations of `SimulatesIntermediateStates` implement the `_simulator_iterator` method. For now it's fine to leave this where it is.

Reply:
Should we mark it deprecated?

Reply:
That would be good - it would empower us to make the breaking change once the Cirq version advances.

Reply:
Added the deprecated attribute. I marked it as v0.11 since that was the highest existing deadline in the project.